# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/util-linux/hotfix-process_vm_readv.patch.xtensa
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- util-linux-2.40.4/misc-utils/lsfd.c.vanilla	2025-04-22 03:42:45.513326514 +0100
+++ util-linux-2.40.4/misc-utils/lsfd.c	2025-04-22 03:49:31.969172796 +0100
@@ -34,10 +34,21 @@
 #include <poll.h>
 #include <sys/select.h>
 
+#define _GNU_SOURCE
 #include <sys/uio.h>
 #include <linux/sched.h>
 #include <sys/syscall.h>
 
+ssize_t process_vm_readv(pid_t pid,
+	const struct iovec *local_iov, unsigned long liovcnt,
+	const struct iovec *remote_iov, unsigned long riovcnt,
+		unsigned long flags)
+
+{
+	return syscall(SYS_process_vm_readv,
+		pid, local_iov, liovcnt, remote_iov, riovcnt, flags);
+}
+
 #ifdef HAVE_LINUX_KCMP_H
 #  include <linux/kcmp.h>
 static int kcmp(pid_t pid1, pid_t pid2, int type,
